src/GoogleSignin.js

```diff
@@ -6,23 +6,28 @@ const { RNGoogleSignin } = NativeModules;
 
 const IS_IOS = Platform.OS === 'ios';
 
-class GoogleSignin {
-  configPromise;
+const NativeModuleLoaded = RNGoogleSignin !== undefined;
 
-  constructor() {
-    if (__DEV__ && !RNGoogleSignin) {
-      console.error(
-        'RN GoogleSignin native module is not correctly linked. Please read the readme, setup and troubleshooting instructions carefully or try manual linking.'
-      );
-    }
+function ensureNativeNoduleLoaded() {
+  if (!NativeModuleLoaded) {
+    throw new Error(
+      'RN GoogleSignin native module is not correctly linked. Please read the readme, setup and troubleshooting instructions carefully or try manual linking.'
+    );
   }
+}
+
+class GoogleSignin {
+  configPromise;
 
   async signIn() {
+    ensureNativeModuleLoaded();
```

Comment:
Would this work if we leave just one `ensureNativeModuleLoaded` inside `GoogleSignin`'s constructor instead of every method?

Reply:
It wouldn't - because there is a global singleton instance which is created on module load, the error then would again be raised upon import of the module instead of only when used.

It possible that something could be done using a proxy (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy), but I am not sure how far support for that goes within react-native - I think the Android runtime is pretty old?